code/modules/antagonists/traitor/equipment/Malf_Modules.dm

```diff
@@ -851,5 +851,119 @@ GLOBAL_LIST_INIT(malf_modules, subtypesof(/datum/ai_module))
 /datum/ai_module/upgrade/mecha_domination/upgrade(mob/living/silicon/ai/AI)
 	AI.can_dominate_mechs = TRUE //Yep. This is all it does. Honk!
 
+
+
+/datum/ai_module/upgrade/voice_changer
+	name = "Voice Changer"
+	description = "Allows you to change the AI's voice. Upgrade is active immediately upon purchase."
+	cost = 40
+	one_purchase = TRUE
+	power_type = /datum/action/innate/ai/voice_changer
+	unlock_text = "<span class='notice'>OTA firmware distribution complete! Voice changer online.</span>"
+	unlock_sound = 'sound/items/rped.ogg'
+
+/datum/action/innate/ai/voice_changer
+	name="Voice Changer"
+	button_icon_state = "doomsday_device"
+	auto_use_uses  = FALSE
+	var/obj/machinery/ai_voicechanger/voice_changer_machine
+
+/datum/action/innate/ai/voice_changer/Activate()
+	if(!voice_changer_machine)
+		voice_changer_machine = new(owner_AI)
+	voice_changer_machine.ui_interact(usr)
+
+/obj/machinery/ai_voicechanger
+	icon = 'icons/obj/machines/nuke_terminal.dmi'
+	name = "Voice Changer"
+	icon_state = "nuclearbomb_base"
+	var/mob/living/silicon/ai/owner
+	var/loudvoice = FALSE
+	var/say_verb
+	var/say_name
+	var/say_span
+	var/changing_voice = FALSE
+	var/prev_loud
+	var/prev_verbs
+	var/prev_span
+
+/obj/machinery/ai_voicechanger/Initialize(mapload)
+	. = ..()
+	if(!isAI(loc))
+		stack_trace("Voice Changer created outside of an AI. Cry at coders")
+		return INITIALIZE_HINT_QDEL
+	owner = loc
+	owner.ai_voicechanger = src
+	prev_verbs = list("say" = "states", "ask" = "queries", "exclaim" = "declares" , "yell" = "alarms"  )
+	prev_span = owner.speech_span
+	say_name = owner.name
+	say_verb = "states"
+	say_span = owner.speech_span
+
+/obj/machinery/ai_voicechanger/ui_interact(mob/user, datum/tgui/ui)
+  ui = SStgui.try_update_ui(user, src, ui)
+  if(!ui)
+    ui = new(user, src, "AiVoiceChanger")
+    ui.open()
+
+
+/obj/machinery/ai_voicechanger/ui_data(mob/user)
+  var/list/data = list()
+  data["voices"] = list() //manually adding this since i dont see other option
+  data["voices"]+=list("Normal")
+  data["voices"]+=list(SPAN_ROBOT)
+  data["voices"]+=list(SPAN_YELL)
+  data["voices"]+=list(SPAN_CLOWN)
```

Comment:
why not just start that list with those instead of adding them 1 by 1

Reply:
I have been infected with catbrain disease. Will fix it